### ISSUE TITLE
Security Vulnerability report

### ISSUE THREAD

**Issue by febinrev** [mate-desktop/mate-desktop#591]
I am a security researcher, and I have found a security vulnerability in the default archive manager of MATE DE. The affected software is Engrampa Archive Manager and the vulnerability is capable of Remote Command Execution upon extracting a crafted Archive.

I would like to safely disclose the details about the vulnerability to MATE devs, please provide me with the right contact information to report the bug.

My Email: febin.sec@gmail.com

Thanks,
Febin

**Comment by lukefromdc** [comment]
One of the team members most familiar with Engrampa's code would be best for this. I haven't done much with Engrampa myself beyond the usual testing of PR's and bugfixes.  If you find a wayland specific vulnerbility anywhere in MATE's new wayland support I will be the one that needs to know.

**Comment by raveit65** [comment]
The right repository is https://github.com/mate-desktop/engrampa
You should post it there.
Or do you mean a non public place to communicate?
Can you make a pull request which fixes the issue?

**Comment by raveit65** [comment]
Discussions in core-team are only accessible by core-team members. I don't want to add a outside collaborator to core-team with all rights for all repositories.
What we can do is to add a new private group `Security-Discussions` which is available for core-team members.  A core-team member can open then a new discussion for a security case and (hopefully) adding the security collaborator to the group.

**Comment by cwendling** [comment]
Yeah, or we could simply discuss it internally after one of us got the info and relay it.  Yet, I can't seem to find the team discussions anymore?  I see a link to some old JSON data, but nothing else… guess I'm missing something?

**Comment by raveit65** [comment]
Github did a redesign to make it better (ha ha) and hide old projects and their discussions. They called projects (classics).
https://github.com/orgs/mate-desktop/projects?type=classic
It's a not easy to find them :-)
You can open now projects or projects (classic).

**Comment by raveit65** [comment]
@cwendling 
Do you have access to https://github.com/orgs/mate-desktop/teams/security-discussions and can you add a collaborator ie. [febinrev](https://github.com/febinrev)

**Comment by cwendling** [comment]
@raveit65 Apparently I don't have enough rights, but I just asked to join.  You probably need to up some rights for me if I should be able to do that.

**Comment by cwendling** [comment]
> Github did a redesign to make it better (ha ha) and hide old projects and their discussions. They called projects (classics).
> https://github.com/orgs/mate-desktop/projects?type=classic
> It's not very intuitive to find them :-)
> You can open now projects or projects (classic).

I see the projects and their related issues, but don't see discussions there?

What I was referring to before was "Team post archive" in the sidebar at https://github.com/orgs/mate-desktop/teams/core-team.  Apparently we're now supposed to use the Discussions feature?  But IIUC we don't have it enabled, and I don't think I can manage that either :)  If Discussions works and has support for private threads, that could work.

**Comment by raveit65** [comment]
> I see the projects and their related issues, but don't see discussions there?

> What I was referring to before was "Team post archive" in the sidebar at https://github.com/orgs/mate-desktop/teams/core-team. Apparently we're now supposed to use the Discussions feature? But IIUC we don't have it enabled, and I don't think I can manage that either :) If Discussions works and has support for private threads, that could work.

My fault, old discussions are archived here https://github.com/orgs/mate-desktop/teams/core-team/members/archived_team_posts.json






> @raveit65 Apparently I don't have enough rights, but I just asked to join. You probably need to up some rights for me if I should be able to do that.

Hmm, i thought all member of a parent group are inside the child group. ........ i need to add all core-teams member by hand :/
@cwendling
You are inside this group now with role:member (for some reasons). Can you add https://github.com/febinrev ?

**Comment by cwendling** [comment]
@raveit65 I can't seem to add people not member of the org…

Anyway, if we had him/her in there (or contacted by email and forwarded), where would we discuss that privately?

**Comment by raveit65** [comment]
@cwendling 
This doesn't work for you?
![grafik](https://github.com/mate-desktop/mate-desktop/assets/961604/c9239a23-4e0e-4006-92f5-2b3c438d9ec4)

Anyway, i need to setup the group new because child group receive notifications of the parent group :/

```
Child teams inherit the parent's access permissions, simplifying permissions management for large groups. Members of child teams also receive notifications when the parent team is @mentioned, simplifying communication with multiple groups of people.
```

**Comment by cwendling** [comment]
> This doesn't work for you?

Nope, I had the UI but everything was grayed-out for non-organization members

**Comment by raveit65** [comment]
You are now in this group as maintainer, try again please.

**Comment by cwendling** [comment]
Nope, still no luck: ![team-add](https://github.com/mate-desktop/mate-desktop/assets/24433571/11982140-6a20-4c0f-8917-895145aaa70d)

**Comment by cwendling** [comment]
@raveit65 what about https://docs.github.com/en/code-security/security-advisories/guidance-on-reporting-and-writing-information-about-vulnerabilities/privately-reporting-a-security-vulnerability?

**Comment by raveit65** [comment]
I enabled discussions and a created a post, but it seems the post is public. Or i missed something.
https://github.com/orgs/mate-desktop/discussions
So this isn't usable for our case.

**Comment by cwendling** [comment]
@raveit65 see above, it seems there's a feature just for this that can be enabled.

**Comment by raveit65** [comment]
Anyway, discussion feature can be used for public discussions or announcements (etc. releases), so it wasn't bad to enable them.
Beside from your given link i am still wondering why it isn't possible to make non public discussions inside a organization.

**Comment by raveit65** [comment]
@febinrev 
You can report the Security Vulnerability here https://github.com/mate-desktop/engrampa/security

@mate-desktop/core-team
 I will enable this for all repos.

**Comment by febinrev** [comment]
> @febinrev You can report the Security Vulnerability here https://github.com/mate-desktop/engrampa/security
> 
> @mate-desktop/core-team I will enable this for all repos.

Just reported the vulnerability there! @raveit65

**Comment by febinrev** [comment]
> We could issue an advisary for now advising disconnection from the Internet prior
to opening any unknown file in Engrampa to block command and control servers.
For high security cases, making Engrampa sure has exited before reconnecting. This
would block most remote exploits. Remember that a local exploit in a downloaded
file BECOMES a remote exploit if the payload opens any network connection.

Disconnection from internet/network would not be an ideal solution for exploits that's coming from crafted archives, most exploits that comes from archives would be Path traversals, buffer overflows and logic flaws. These attacks can't be prevented by internet disconnection/isolation. 

The vulnerability I just reported is such a vulnerability, in which the crafted archive writes some file/script to somewhere and that file will be executed after some time, probably triggered after a reboot/relogin.

**Comment by lukefromdc** [comment]
That means such archives should not be opened at all, and until this is fixed archives not from a trusted source have to be treated as though they were attachments in an untrusted email: open only from a live USB stick system. Royal nuisance, few will do this, so we cannot recommend any mitigation people will actually use until this is fixed.

**Comment by raveit65** [comment]
Closing this. Discussions should be done in Security Vulnerability report at https://github.com/mate-desktop/engrampa/security/advisories
